src/utils/address_utils.ts

```diff
@@ -17,3 +20,23 @@ export const objectETHAddressNormalizer = <T>(obj: T) => {
         ...normalized,
     };
 };
+
+/**
+ * Attempts to find the address of the token and throws if not found
+ *
+ * @param address the uppercase symbol of the token (ex. `REP`) or the address of the contract
+ * @param chainId the Network where the address should be hosted on.
+ */
+export function findTokenAddressOrThrowApiError(address: string, field: string, chainId: ChainId): string {
+    try {
+        return findTokenAddressOrThrow(address, chainId);
+    } catch (e) {
+        throw new ValidationError([
+            {
+                field,
+                code: ValidationErrorCodes.ValueOutOfRange,
```

Comment:
Is this the most appropriate code? 🤔  How about `TokenNotSupported` or `InvalidField`?

Reply:
ya agree. i didn't love the previous error either, changing